java/src/org/openqa/selenium/grid/node/local/SessionSlot.java

```diff
@@ -144,7 +144,16 @@ public Either<WebDriverException, ActiveSession> apply(CreateSessionRequest sess
     }
 
     try {
-      Either<WebDriverException, ActiveSession> possibleSession = factory.apply(sessionRequest);
+      Capabilities mergedCapabilities =
+        sessionRequest.getDesiredCapabilities().merge(this.getStereotype());
```

Comment:
Would this overwrite the browser options passed by the user? Like, if the custom stereotype has a `goog:chromeOptions`, would it overwrite what the user has passed?

Reply:
I think so. The capabilities of the slot override the capabilities of the incoming session request in case of custom driver-configuration.

Reply:
I think that is not ideal, because we would not honor the preferences set by the user. If we want to do this, probable this needs a deep merge of browser specific options. What do you think?

Reply:
Right, but if a user wants to run headless, and the stereotype has a setting to set the browser language. Both things would live inside `goog:chromeOptions`. I think having the stereotype ones take precedence is unexpected and would break the behavior users currently have in Grid. 

The current merge logic replaces the value at the top level, if we want to do this, we should follow the same logic for browser preferences, like `goog:chromeOptions`, `moz:firefoxOptions`, etc...

Reply:
Ahh okay, I think I see your point. I should have been clearer, I meant the merge logic for browser preferences. Let me see how we can achieve that or what it will take to make it possible.